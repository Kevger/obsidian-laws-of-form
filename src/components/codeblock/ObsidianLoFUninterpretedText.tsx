/**
 * React component for displaying uninterpreted text in a code block.
 *
 * @param text - The raw text to display.
 * @returns - JSX.Element
 */
export function ObsidianLoFUninterpretedText({
  text,
  placeRight,
}: {
  text: string;
  placeRight: boolean;
}): JSX.Element {
  return (
    <span
      style={{ marginLeft: placeRight ? "auto" : undefined, whiteSpace: "pre" }}
    >
      {text}
    </span>
  );
}
