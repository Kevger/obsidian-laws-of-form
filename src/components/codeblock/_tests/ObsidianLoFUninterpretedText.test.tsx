// write tests for ObsidianLoFUninterpretedText here
import { type ReactTestRenderer, create as render } from "react-test-renderer";
import { ObsidianLoFUninterpretedText } from "../ObsidianLoFUninterpretedText";

describe("ObsidianLoFUninterpretedText", () => {
  describe("given we have an uninterpreted text", () => {
    const uninterpretedText =
      "(Lorem ipsum dolor () sit amet, consectetur adipiscing elit.)";
    let component: ReactTestRenderer;

    describe("and it's placed left", () => {
      beforeEach(() => {
        component = render(
          <ObsidianLoFUninterpretedText
            text={uninterpretedText}
            placeRight={false}
          />
        );
      });
      it("renders correctly", () => {
        const snapshot = component.toJSON();
        expect(snapshot).toMatchSnapshot();
      });

      it("contains the uninterpreted text", () => {
        expect(component.root.findByType("span").children[0]).toBe(
          uninterpretedText
        );
      });
    });

    describe("and it's placed right", () => {
      beforeEach(() => {
        component = render(
          <ObsidianLoFUninterpretedText
            text={uninterpretedText}
            placeRight={true}
          />
        );
      });
      it("renders correctly", () => {
        const snapshot = component.toJSON();
        expect(snapshot).toMatchSnapshot();
      });

      it("contains the uninterpreted text", () => {
        expect(component.root.findByType("span").children[0]).toBe(
          uninterpretedText
        );
      });
    });
  });
});
