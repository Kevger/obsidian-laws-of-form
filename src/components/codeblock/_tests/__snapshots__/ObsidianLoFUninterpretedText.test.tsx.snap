// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObsidianLoFUninterpretedText given we have an uninterpreted text and it's placed left renders correctly 1`] = `
<span
  style={
    {
      "marginLeft": undefined,
      "whiteSpace": "pre",
    }
  }
>
  (Lorem ipsum dolor () sit amet, consectetur adipiscing elit.)
</span>
`;

exports[`ObsidianLoFUninterpretedText given we have an uninterpreted text and it's placed right renders correctly 1`] = `
<span
  style={
    {
      "marginLeft": "auto",
      "whiteSpace": "pre",
    }
  }
>
  (Lorem ipsum dolor () sit amet, consectetur adipiscing elit.)
</span>
`;
